rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java

```diff
@@ -0,0 +1,316 @@
+		proxyUser = new Text(container, SWT.BORDER);
+		proxyUser.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		proxyUser.setText(proxyUserSetting);
+		proxyUser.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				proxyUserSetting = proxyUser.getText();
+				updateNetworkState();
+			}
+		});
+		new Label(container, SWT.NORMAL).setText("Proxy password:");
+		proxyPwd = new Text(container, SWT.BORDER);
+		proxyPwd.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		proxyPwd.setText(proxyPwdSetting);
+		proxyPwd.setEchoChar('*');
+		proxyPwd.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				proxyPwdSetting = proxyPwd.getText();
+				updateNetworkState();
+			}
+		});
+		new Label(container, SWT.NORMAL).setText("No Proxy for:");
+		noProxyHosts = new Text(container, SWT.BORDER);
+		noProxyHosts.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		noProxyHosts.setText(noProxyHostsSetting);
+		noProxyHosts.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				noProxyHostsSetting = noProxyHosts.getText();
+				updateNetworkState();
+			}
+		});
+	}
+
+	private void createMavenSettingsWidgets(final Composite container) {
+		new Label(container, SWT.NORMAL).setText("Path to maven settings:");
+		Composite mavenPathControl = new Composite(container, SWT.NORMAL);
+		mavenPathControl.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		mavenPathControl.setLayout(new GridLayout(2, false));
+		pathToMavenSettingsFile = new Text(mavenPathControl, SWT.BORDER);
+		pathToMavenSettingsFile.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		pathToMavenSettingsFile.setText(pathToMavenSettingsFileSettings);
+		pathToMavenSettingsFile.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				pathToMavenSettingsFileSettings = pathToMavenSettingsFile.getText();
+			}
+		});
+		Button fileSelection = new Button(mavenPathControl, SWT.BORDER);
+		fileSelection.setText("...");
+		fileSelection.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				FileDialog fileDialog = new FileDialog(container.getShell());
+				fileDialog.setText("Select Maven settings");
+				String path = fileDialog.open();
+				if (path != null) {
+					pathToMavenSettingsFile.setText(path);
+				}
+			}
+		});
+	}
+
+	private void updateNetworkState() {
+		Runnable runnable = new Runnable() {
+
+			@Override
+			public void run() {
+				if (isInternetAvailable(true)) {
+					if (workOffline) {
+						updateNetworkSateWidget("work offline", SWT.COLOR_DARK_YELLOW);
+					} else {
+						updateNetworkSateWidget("connected", SWT.COLOR_DARK_GREEN);
+					}
+				} else {
+					updateNetworkSateWidget("diconnected", SWT.COLOR_RED);
+				}
+			}
+		};
+		new Thread(runnable).start();
```

Comment:
Why do we need to run this in a separate thread? The called method calls `Display.asyncExec` anyway.